### PR TITLE
Remove RelativeLayout from Database quickstart

### PR DESCRIPTION
**Pull request by samtstern**
See #1217 

I kept all the layouts the same except I changed a lot of the padding/margin values to be the standard 8/16/24 scale instead of the 5/10/15 that was being used in this app.

**activity_main**

| Before | After |
|--------| ------|
|  Screen Shot 2021-01-12 at 11 39 58 AM |  Screen Shot 2021-01-12 at 11 43 28 AM |

**activity_new_post**

| Before | After |
|-------- | ------|
| Screen Shot 2021-01-12 at 11 48 03 AM | Screen Shot 2021-01-12 at 11 50 52 AM |

**activity_post_detail**

| Before | After |
|-------- | ------|
| Screen Shot 2021-01-12 at 11 52 43 AM | Screen Shot 2021-01-12 at 12 01 46 PM|

**activity_sign_in**

| Before | After |
|-------- | ------|
| Screen Shot 2021-01-12 at 12 04 34 PM | Screen Shot 2021-01-12 at 12 12 21 PM |

**fragment_all_posts**

N/A this is a FrameLayout

**include_post_author**

| Before | After |
|-------- | ------|
| Screen Shot 2021-01-12 at 12 15 07 PM | Screen Shot 2021-01-12 at 12 16 56 PM |

**include_post_text**

| Before | After |
|-------- | ------|
| Screen Shot 2021-01-12 at 12 17 56 PM | Screen Shot 2021-01-12 at 12 19 10 PM |

**item_comment**

| Before | After |
|-------- | ------|
| Screen Shot 2021-01-12 at 12 20 32 PM | Screen Shot 2021-01-12 at 12 24 16 PM|

**item_post**

| Before | After |
|-------- | ------|
| Screen Shot 2021-01-12 at 12 25 10 PM | Screen Shot 2021-01-12 at 12 29 39 PM |

### CODE REVIEW

**Comment by samtstern** [comment]
@rosariopfernandes I can't assign you directly but please review this one when you get a chance!

File: database/app/src/main/res/layout/activity_sign_in.xml [hunk 1]
```diff
@@ -1,86 +1,86 @@
...
-        <Button
-            android:id="@+id/buttonSignIn"
-            android:layout_marginStart="8dp"
-            android:layout_marginEnd="8dp"
-            android:layout_width="150dp"
-            android:layout_height="wrap_content"
-            android:text="@string/sign_in" />
+        android:layout_marginTop="16dp"
+        android:ellipsize="end"
+        android:gravity="center_horizontal"
```
Comment (rosariopfernandes) [inline database/app/src/main/res/layout/activity_sign_in.xml:40]: I believe we no longer need this `gravity` attribute

File: database/app/src/main/res/layout/activity_sign_in.xml [hunk 2]
```diff
@@ -1,86 +1,86 @@
...
+        app:layout_constraintHorizontal_bias="0.5"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/icon" />
 
+    <EditText
+        android:id="@+id/fieldPassword"
+        android:layout_width="150dp"
+        android:layout_height="wrap_content"
+        android:ellipsize="end"
+        android:gravity="center_horizontal"
```
Comment (rosariopfernandes) [inline database/app/src/main/res/layout/activity_sign_in.xml:55]: Same comment as above

File: database/app/src/main/res/layout/item_post.xml [hunk 3]
```diff
@@ -16,44 +17,41 @@
...
+            app:layout_constraintTop_toTopOf="@+id/star"
+            tools:text="7" />
 
-        <include layout="@layout/include_post_text"
+        <include
+            layout="@layout/include_post_text"
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
-            android:layout_alignParentLeft="true"
             android:layout_below="@+id/postAuthorLayout"
```
Comment (rosariopfernandes) [inline database/app/src/main/res/layout/item_post.xml:47]: One more attribute that could removed: `layout_below`

File: database/app/src/main/res/layout/item_post.xml [hunk 4]
```diff
@@ -16,44 +17,41 @@
...
+            android:src="@drawable/ic_toggle_star_outline_24"
+            app:layout_constraintBottom_toBottomOf="@+id/postAuthorLayout"
+            app:layout_constraintEnd_toStartOf="@+id/postNumStars"
+            app:layout_constraintTop_toTopOf="@+id/postAuthorLayout" />
+
+        <TextView
+            android:id="@+id/postNumStars"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:gravity="center"
```
Comment (rosariopfernandes) [inline database/app/src/main/res/layout/item_post.xml:37]: I think we can remove this one too

File: database/app/src/main/res/layout/activity_post_detail.xml [hunk 5]
```diff
@@ -1,62 +1,66 @@
...
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
 
     <include
         android:id="@+id/postTextLayout"
         layout="@layout/include_post_text"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:layout_alignParentLeft="true"
         android:layout_below="@+id/postAuthorLayout"
```
Comment (rosariopfernandes) [inline database/app/src/main/res/layout/activity_post_detail.xml:25]:
```suggestion
```
Can be removed

File: database/app/src/main/res/layout/activity_post_detail.xml [hunk 6]
```diff
@@ -1,62 +1,66 @@
...
 
     <androidx.recyclerview.widget.RecyclerView
         android:id="@+id/recyclerPostComments"
         android:layout_width="match_parent"
-        android:layout_height="match_parent"
-        android:layout_below="@+id/commentForm"
+        android:layout_height="0dp"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/buttonPostComment"
+        tools:layout_editor_absoluteX="16dp"
```
Comment (rosariopfernandes) [inline database/app/src/main/res/layout/activity_post_detail.xml:63]: Does `match_parent` work with ConstraintLayout?
I wonder if this `tools` attribute is what's making the RecyclerView look ok on the IDE preview.

I'd suggest adding horizontal constraints to it and changing the width to `0dp`

**Review by rosariopfernandes — COMMENTED** [review]
@samtstern I like how you used tables for the Before and After on the PR description.

These changes look really good, but I have added some comments

**Comment by samtstern** [comment]
@rosariopfernandes thanks for the quick comments!  I think I addressed them now.

**Review by rosariopfernandes — APPROVED** [review]
LGTM!

**Pull request merged**